### DISCUSSION TITLE
Is stream_queue featur flag enabled by an implicit dependency or requirement?

### DISCUSSION THREAD

**Discussion by dreiucker** [rabbitmq/rabbitmq-server#7092] [Questions] [answered]
Hello,

I'm currently setting up RabbitMQ 3.11.2 on Erlang 25.0.1. Yes, I knew there are newer version. What I'm currently wondering is about which feature flags to use with the new version. I know there are requirements to have specific feature flags enabled for 3.11 https://blog.rabbitmq.com/posts/2022/07/required-feature-flags-in-rabbitmq-3.11/ 

I enabled all required feature flags and most of the others, expect the ones for stream:

`RABBITMQ_FEATURE_FLAGS=quorum_queue,implicit_default_bindings,virtual_host_metadata,maintenance_mode_status,user_limits,classic_mirrored_queue_version,classic_queue_type_delivery_support,direct_exchange_routing_v2,drop_unroutable_metric,empty_basic_get_metric,feature_flags_v2,listener_records_in_ets,tracking_records_in_ets`

However, when I check the feature flags via CLI stream_queue are enabled:
```
rabbitmqctl list_feature_flags
Listing feature flags ...
name	state
classic_mirrored_queue_version	enabled
classic_queue_type_delivery_support	enabled
direct_exchange_routing_v2	enabled
drop_unroutable_metric	enabled
empty_basic_get_metric	enabled
feature_flags_v2	enabled
implicit_default_bindings	enabled
listener_records_in_ets	enabled
maintenance_mode_status	enabled
quorum_queue	enabled
stream_queue	enabled
stream_single_active_consumer	disabled
tracking_records_in_ets	enabled
user_limits	enabled
virtual_host_metadata	enabled
```

Have I overlooked a requirement or implicit dependency dependency that introduces stream_queues as default? 

Looking forward to get an explanation for that behavior. Thank you :)

**Comment by kjnilsson** [comment, marked as answer]
Feature flags are not meant to be disabled or used as a configuration
system. They are purely there to allow no downtime upgrades. Yes most
likely one of the enabled feature flags depended on stream_queue.

https://www.rabbitmq.com/feature-flags.html

On Mon, 30 Jan 2023 at 10:00, Monika-Lydia Dreiucker <
***@***.***> wrote:

> Hello,
>
> I'm currently setting up RabbitMQ 3.11.2 on Erlang 25.0.1. Yes, I knew
> there are newer version. What I'm currently wondering is about which
> feature flags to use with the new version. I know there are requirements to
> have specific feature flags enabled for 3.11
> https://blog.rabbitmq.com/posts/2022/07/required-feature-flags-in-rabbitmq-3.11/
>
> I enabled all required feature flags and most of the others, expect the
> ones for stream:
>
>
> RABBITMQ_FEATURE_FLAGS=quorum_queue,implicit_default_bindings,virtual_host_metadata,maintenance_mode_status,user_limits,classic_mirrored_queue_version,classic_queue_type_delivery_support,direct_exchange_routing_v2,drop_unroutable_metric,empty_basic_get_metric,feature_flags_v2,listener_records_in_ets,tracking_records_in_ets
>
> However, when I check the feature flags via CLI stream_queue are enabled:
>
> rabbitmqctl list_feature_flags
> Listing feature flags ...
> name	state
> classic_mirrored_queue_version	enabled
> classic_queue_type_delivery_support	enabled
> direct_exchange_routing_v2	enabled
> drop_unroutable_metric	enabled
> empty_basic_get_metric	enabled
> feature_flags_v2	enabled
> implicit_default_bindings	enabled
> listener_records_in_ets	enabled
> maintenance_mode_status	enabled
> quorum_queue	enabled
> stream_queue	enabled
> stream_single_active_consumer	disabled
> tracking_records_in_ets	enabled
> user_limits	enabled
> virtual_host_metadata	enabled
>
> Have I overlooked a requirement or implicit dependency dependency that
> introduces stream_queues as default?
>
> Looking forward to get an explanation for that behavior. Thank you :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/discussions/7092>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJAHFG4KKOIYCOYJFGFZFTWU6GLJANCNFSM6AAAAAAUK55MIA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
*Karl Nilsson*
